### PR TITLE
fix: error when choosing no for experts (CMC-1375)

### PR DESCRIPTION
**Pull request by pd-dudkiewicz**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/CMC-1375

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/user/RespondToDefenceCallbackHandler.java [hunk 1]
```diff
@@ -71,6 +73,19 @@ private CallbackResponse validateUnavailableDates(CallbackParams callbackParams)
             .build();
     }
 
+    private CallbackResponse validateExperts(CallbackParams callbackParams) {
```
Comment (sabahirfan) [inline src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/user/RespondToDefenceCallbackHandler.java]: can we pull this duplicate method to common place and reuse? We can pass a `DQ` reference to that common method.

**Review by sabahirfan — DISMISSED** [review]
one small suggestion, otherwise LGTM

File: src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/user/RespondToClaimCallbackHandler.java [hunk 2]
```diff
@@ -77,6 +80,19 @@ private CallbackResponse validateUnavailableDates(CallbackParams callbackParams)
             .build();
     }
 
+    private CallbackResponse validateExperts(CallbackParams callbackParams) {
+        CaseData caseData = callbackParams.getCaseData();
+        var experts = caseData.getRespondent1DQ().getExperts();
+        List<String> errors = new ArrayList<>();
+        if (experts.getExpertRequired() == YES && experts.getDetails() == null) {
```
Comment (swalker125) [inline src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/user/RespondToClaimCallbackHandler.java]: thoughts on making this a method in Experts class?

File: src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/user/RespondToClaimCallbackHandler.java [hunk 3]
```diff
@@ -77,6 +80,19 @@ private CallbackResponse validateUnavailableDates(CallbackParams callbackParams)
             .build();
     }
 
+    private CallbackResponse validateExperts(CallbackParams callbackParams) {
```
Comment (swalker125) [inline src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/user/RespondToClaimCallbackHandler.java]: There is an example in create claim where we have a ParticularsOfClaimValidator interface. Could we implement that in the two callback handlers to just have one place where this method has to exist?

File: src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/user/RespondToClaimCallbackHandler.java (hunk 2 above)
Comment (pd-dudkiewicz) [reply src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/user/RespondToClaimCallbackHandler.java]: it's simple enough I'd keep it in validator

File: src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/user/RespondToClaimCallbackHandler.java (hunk 3 above)
Comment (pd-dudkiewicz) [reply src/main/java/uk/gov/hmcts/reform/unspec/handler/callback/user/RespondToClaimCallbackHandler.java]: done 👍

**Review by dharmendrak — APPROVED** [review]

**Pull request closed without merging**